### DISCUSSION TITLE
Formattering av info

### DISCUSSION THREAD

**Discussion by kennidenni** [fremtind/jokul#2530] [Annet]
Heisann! 

Vi som jobber på kundeklienten har et behov for litt forskjellige formatters og regner nesten med at andre også har behov for det samme. 
Det kan være fint om dato, summer, telefonnummer og annet er formattert på likt i alle løsninger. 

Her kan dere se summer som er på format: 1 100 000 kr
![image](https://user-images.githubusercontent.com/31621649/142985468-e51be503-01d2-4902-b1a7-ab3149a6a2cc.png)

Her kan dere se lenge og dato: 
Lengde: 200 000 km
Dato: 10.09.2021
![image](https://user-images.githubusercontent.com/31621649/142985344-a4dc5cea-c3dd-4aab-8b87-54cefffabc4a.png)

Andre ting kan være tlf på dette formatet: 
Telefon: 999 99 999

Jeg kan raskt flytte over tlf og dato til Jøkul fra vår løsning, men for sum (kr) og lengde (km) bruker vi @sb1/ffe-formatters. Tipper at vi ville lagd disse på nytt fra bunnen dersom vi skal ha de hos oss for å slippe den avhengigheten.

**Comment by wkillerud** [comment]
Ser _de_ igjen bruker `underscore.string`. Tenker vi dropper mellomleddet ja.

Ser `constants` og `validators` har postfixet `-utils`, så vi bør vel legge oss på samme linje for denne nye pakka.

**Reply by piofinn** [reply]
Her bør vi vel sørge for å bruke innebygde APIer så langt som mulig. [`Intl.NumberFormat()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat) har et vell av muligheter og støttes av alle moderne nettlesere. Da har vi også muligheten til å bestemme osss for om vi alltid skal bruke norsk tallformat eller skal tilpasse oss brukerens lokalisasjonsinnstillinger.

**Comment by onordbo** [comment]
Er jeg helt på bærtur om jeg tenker denne burde sees i sammenheng med #2454? 🤔

**Reply by wkillerud** [reply]
I utgangspunktet snakker vi bare visning av data fra systemet og ikke formateringen av brukerens input i inputfeltet. Vi kan utforske [input masks](https://coreui.io/docs/2.1/components/masked-input/), men før vi gjør noe sånt bør vi ha tenkt oss om.

**Reply by kennidenni** [reply]
Hmm ja, her mener jeg bare formattering av verdier på siden i løsningene.

**Reply by piofinn** [reply]
Input masks kan være en god idé på lange numre som kontonummer, kredittkortnummer eller fødsels- og personnummer, men jeg er enig i at det bør være gjort en tankerunde først. Uansett bør jo reglene som lages her fint kunne brukes direkte for å lage slike masker om ønskelig 😊

**Comment by ghost** [comment]
Formattering av personnummer/ssn hadde også vært veldig nyttig, og sikkert noe som mange vil ha behov for. Gjerne all formattering uten space slik at copy-paste ikke får med mellomrom.

**Comment by piofinn** [comment]
Hepp! Laget en draft pull request med det jeg har drodlet på så langt hvis dere vil ta en titt 😊:

https://github.com/fremtind/jokul/pull/2655
